Add container arch to system info

agners opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
This is a follow-up for #146537 and #146561. It removed aiofiles from the core requirements as suggested in https://github.com/home-assistant/core/pull/146561#issuecomment-2963862553. It also makes the detected architecture available through system info. The container architecture can be different from the reported CPU architecture in cases where the underlying OS is capable of running other architectures (typically x86 on x86_64 or armv7 on aarch64 operating systems).

For Supervisor based installations it also uses the architecture already available from Supervisor. This makes the information more coherent and avoids unnecessary code duplication.

<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 
- Link to developer documentation pull request: 
- Link to frontend pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

agners committed:
Load container architecture in system info

Make the Alpine Linux container architecture from the Alpine package
manager available in the system info. This information is useful not
only to create the deprecation repairs, but also as a general
system information since the container architecture may differ from
the host architecture, for example, when running a 64-bit host with
a 32-bit container.

agners committed:
Use Supervisor arch to determine container arch

For installations with Supervisor, use the architecture from the
Supervisor API instead of reading the architecture from the
container. This is more coherent as we trust the Supervisor data
already, and gets rid of an unnecessary dependency for aiofiles.

agners committed:
Revert "Add aiofiles to pyproject.toml (#146561)"

This reverts commit fb4c77d43b3e45e22cbd075d894459e9970e0587.

aiofiles is no longer needed by the homeassistant integration.

joostlek reviewed (changes requested):
Tests are failing, can you take a look?

agners committed:
Fix cloud test

agners committed:
Drop unnecessary pytest parameter

agners committed:
Delay deprecation checks until Home Assistant is fully initialized

This aims to delay the async_get_system_info() call until the hassio
integration is fully initialized, to make sure the installation method
is determined correctly.

agners commented:
> Tests are failing, can you take a look?

Hm, that was harder than I thought, and it actually highlighted a bug I was seeing here too: On a Supervised installation which also has a deprecated arch, not only the `hassio` deprecation gets created, but also the `homeassistant` deprecation gets created. The reason is that `async_get_system_info()` returns the Container installation method, because at that time the `hassio` integration is not setup yet, so the installation method is not detected correctly :cry: 

So this lead to the tests failing due to too many issues, since the `homeassistant` integration creates an issue too.

I've come up with a solution which delays the deprecation issue creation until Home Assistant is up and running. This correctly returns the installation method for Supervisor based installations when the deprecation issue is created in the `homeassistant` integration.

agners committed:
Fire started event to fix pytests
